source/val/validate_memory.cpp

```diff
@@ -2115,16 +2115,21 @@ spv_result_t ValidateCooperativeMatrixLoadStoreKHR(ValidationState_t& _,
 
   const auto layout_index =
       (inst->opcode() == spv::Op::OpCooperativeMatrixLoadKHR) ? 3u : 2u;
-  const auto colmajor_id = inst->GetOperandAs<uint32_t>(layout_index);
-  const auto colmajor = _.FindDef(colmajor_id);
-  if (!colmajor || !_.IsIntScalarType(colmajor->type_id()) ||
-      !(spvOpcodeIsConstant(colmajor->opcode()) ||
-        spvOpcodeIsSpecConstant(colmajor->opcode()))) {
+  const auto layout_id = inst->GetOperandAs<uint32_t>(layout_index);
+  const auto layout_inst = _.FindDef(layout_id);
+  uint64_t layout;
+  if (!layout_inst || !_.IsIntScalarType(layout_inst->type_id()) ||
+      !(spvOpcodeIsConstant(layout_inst->opcode())) ||
+      !_.EvalConstantValUint64(layout_id, &layout)) {
```

Comment:
This function returns false if the constant is a spec constant. So this change has unintentionally disallowed spec constants. This call seems unnecessary in this location though. It could be moved below.

Please add a testcase to cover this case.

Reply:
Oh yes, I keep forgetting that the behaviour of this function was changed (I can understand why folks might have wanted to do that but that's something I think I'd like to discuss too at some point; separate discussion though).

>Please add a testcase to cover this case.

Will do.